### PR TITLE
contributing: updates for Mbed OS 6.15

### PR DESCRIPTION
**Pull request by 0xc0170**
Reviewing what we have and updating the information for multiple pages (separate commits per each change).

### CODE REVIEW

**Comment by ChrisGooch-Arm** [comment]
Hi @0xc0170 - do we want to merge these changes for 6.15?

**Review by donatieng — APPROVED** [review]

File: docs/contributing/contributing.md [hunk 1]
```diff
@@ -1,5 +1,3 @@
 # Contributing overview
 
-The Arm Mbed OS porting guide describes how to a add development board Mbed OS and configure it. It includes porting information about related features, such as cellular, exporters and Mbed TLS. It also contains information about CMSIS, Keil RTX and Thread safety.
```
Comment (ChrisGooch-Arm) [inline docs/contributing/contributing.md:3]: this file isn't actually used as far as I can tell (not in docs.json)

File: docs/contributing/guidelines/workflow.md [hunk 2]
```diff
@@ -96,7 +96,6 @@ The following template is automatically provided when you raise a pull request a
     ## Reviewers (*optional*)
 
     ----------------------------------------------------------------------------------------------------------------
-    ## Release Notes (*required for feature/major PRs*)
```
Comment (ChrisGooch-Arm) [inline docs/contributing/guidelines/workflow.md:99]: not sure why this is being removed from the template? the explanatory text further down still has a section on release notes

File: docs/contributing/contributing.md (hunk 1 above)
Comment (0xc0170) [reply docs/contributing/contributing.md:3]: I can remove the file, or ?

File: docs/contributing/guidelines/workflow.md (hunk 2 above)
Comment (0xc0170) [reply docs/contributing/guidelines/workflow.md:99]: We removed release notes section from the PR template. the description is used as release notes.

**Comment by 0xc0170** [comment]
>Hi @0xc0170 - do we want to merge these changes for 6.15?

Yes, we shall merge this when we can

File: docs/contributing/contributing.md (hunk 1 above)
Comment (ChrisGooch-Arm) [reply docs/contributing/contributing.md:3]: I can tidy this later on. If the file is used then the edit looks sensible in any case, so happy to accept and merge this PR.

**Pull request merged**